src/index.ts

```diff
@@ -73,7 +74,21 @@ export const generate = <A>(options: GenerateOptions<A>): TaskEither<unknown, vo
 				either.chain(resolved => reportIfFailed(decoder.decode(resolved))),
 			);
 
-		await write(out, getUnsafe(options.language({ resolveRef })(specs)));
+		const deepLookup: DeepLookup = (node: unknown, codec, refCodec) =>
```

Comment:
I'm not sure this should be a parameter to language function. It looks more like a common helper as it's fully based on `resolveRef`